src/sage/graphs/graph.py

```diff
@@ -4328,6 +4328,228 @@ def weight(x):
+        some vertex `u` with `v` being its `M`-matched neighbor, each neighbor
+        of `v` is reachable from `u` through an even length `M`-alternating
+        path as described above::
+
+            sage: # K(4) ☉ K(3, 3) is matching covered
+            sage: G = Graph()
+            sage: G.add_edges([
+            ....:    (0, 2), (0, 3), (0, 4), (1, 2),
+            ....:    (1, 3), (1, 4), (2, 5), (3, 6),
+            ....:    (4, 7), (5, 6), (5, 7), (6, 7)
+            ....: ])
+            sage: M = Graph(G.matching())
+            sage: G.is_matching_covered(M)
+            True
+            sage: u = 0
+            sage: v = next(M.neighbor_iterator(u))
+            sage: S = G.M_alternating_even_mark(u, M)
+            sage: (set(G.neighbor_iterator(v))).issubset(S)
+            True
+
+        For a bicritical graph `G` with a perfect matching `M` and for some
+        vertex `u` with its `M`-matched neighbor being `v`, each vertex of the
+        graph distinct from `v` is reachable from `u` through an even length
+        `M`-alternating path as described above::
+
+            sage: # Petersen graph is bicritical
+            sage: G = graphs.PetersenGraph()
+            sage: M = Graph(G.matching())
+            sage: G.is_bicritical(M)
+            True
+            sage: u = random.choice(G.vertices())  # needs random
+            sage: v = next(M.neighbor_iterator(u))
+            sage: S = G.M_alternating_even_mark(u, M)
+            sage: S == (set(G.vertices()) - {v})
+            True
+
+        TESTS:
+
+        Giving a wrong vertex::
+
+            sage: G = graphs.HexahedralGraph()
+            sage: M = G.matching()
+            sage: u = G.order()
+            sage: S = G.M_alternating_even_mark(u, M)
+            Traceback (most recent call last):
+            ...
+            ValueError: '8' is not a vertex of the graph
+
+        Giving a wrong matching::
+
+            sage: G = graphs.CompleteGraph(6)
+            sage: M = [(0, 1), (0, 2)]
+            sage: u = 0
+            sage: S = G.M_alternating_even_mark(u, M)
+            Traceback (most recent call last):
+            ...
+            ValueError: the input is not a matching
+            sage: G = graphs.CompleteBipartiteGraph(3, 3)
+            sage: M = [(2*i, 2*i + 1) for i in range(4)]
+            sage: u = 0
+            sage: S = G.M_alternating_even_mark(u, M)
+            Traceback (most recent call last):
+            ...
+            ValueError: the input is not a matching of the graph
+
+        REFERENCES:
+
+        - [LR2004]_
+
+        .. SEEALSO::
+            :meth:`~sage.graphs.graph.Graph.is_factor_critical`,
+            :meth:`~sage.graphs.graph.Graph.is_matching_covered`,
+            :meth:`~sage.graphs.graph.Graph.is_bicritical`
+        """
+
+        G = self.to_simple()
+
+        # The input vertex must be a valid vertex of the graph
+        if vertex not in G.vertices():
```

Comment:
use `if vertex not in self:`. the call to `G.vertices()` returns a list of vertices and there is no need to build this list.